README.buildsys.md

```diff
@@ -0,0 +1,207 @@
+# The GAP build system
+
+This file is meant to give an overview of how the GAP build system works. It
+is targeted at people who need to work on the build system itself (to extend
+it, fixes bugs in it, etc.). It should not be necessary to read this if all
+you want to do is compile GAP and work on the GAP library and kernel.
+
+Note that this really is just an overview; for details, please refer to the
+comments inside the various parts of the build system.
+
+
+## Prerequisites
+
+In order to work on the build system, you need at least the following:
+
+* GNU autoconf (we recommend 2.69 or later)
+* GNU libtool (we recommend 2.4.2 or later)
```

Comment:
GMP, Boehm GC and libatomic_ops all use automake (note the presence of `Makefile.am` files). But this is irrelevant, as we don't need to run (the equivalent of) `autogen.sh` for them.